### ISSUE TITLE
[RFC][SearchBundle] Request-based range filter

### ISSUE THREAD

**Issue by okwinza** [Sylius/Sylius#4319]
IMO current implementation of facet-based range filter for attributes like price is... not so good.
Moreover it is pretty common to have such filter for price for example. Customer should be able to specify exact GTE and LTE values to narrow the search.

I'm ready to tackle this issue myself, however, it's is not completely clear for me what is the best approach would be. 

I appreciate any feedback.

**Comment by Niiko** [comment]
Yes, i agree too.

We can take examples on ldlc.com or amazon.com (two different approches) : 
- with a slider with two ranges (between max and min)
- with facets like actual and a subform between two amounts 

The second approach has the benefits of "showing" how much results on each facet.

We can also imagine only the "subform".

On facets, we can also improve concept of distribution. For now, they are configured in config.yml, but price filter can differ a lot between two taxons.

I can help too on this one.

**Comment by peteward** [comment]
We have the slider implementation for price using Elasticsearch.

There's a few ways that this could be implemented in Sylius core, but probably needs to be triggered somehow by the facet-type. 

checkboxes for multiple options get submitted in the request as a numerical array, so you could implement a range as:

```
filters[price][low] = 10
filters[price][high] = 100
```

Then check for these keys when processing the submitted filters against a `stats` type of facet.

You then need a template/client-side implementation that renders the form as you want it and populates this form appropriately.

I'm not sure about the MySQL implementation but for ES if you use a `stats` type aggregation it provides you with the low and high bounds of values to render in a slider or input boxes as you prefer.

It's been a while since I look at the sylius config since we overwrote it but the default types here are:

```
        facets:
            taxons:
                display_name: 'Basic categories'
                type: terms
                value: ~
            price:
                display_name: 'Available prices'
                type: range
                values:
                    - { from: 0, to: 2000}
                    - { from: 2001, to: 5000}
                    - { from: 5001, to: 10000}
```

It would be good to offer a `stats` option or similar as described above - it is easy to implement with ES as an aggregation and filter, but would need some work to make it compatible with MySQL.

**Comment by okwinza** [comment]
much thanks for your input @peteward, much appreciated!

**Comment by gabiudrescu** [comment]
we implemented price filters like this: we have an application wide setting that states how many steps the price range filter should have - it defaults to 4 right now. we plan to make this available for each taxonomy at a later point.

when the user goes to /listing, all product prices are divided in 4 relatively equal ranges and shown to the user as links, so that he can choose the most relevant for him.

also, in case he has some specific needs, he has two input boxes to use for exact match.

we got inspired by the behavior of amazon.com for this and we rely on elasticSearch.

the idea of using sliders could sound good for mobile usage, but for desktop is a bit cumbersome, if you have big ranges of prices - e.g. between 10 and 1000 € products like watches. it's pretty hard too match that exact 200€ budget limit.

and from the usability point of view, it's easier to click one link or fill in two boxes rather than use a slider. I remember I read an article on that a few years ago, but I can't find the link right now.
